system/HTTP/CURLRequest.php

```diff
@@ -479,10 +479,14 @@ protected function setResponseHeaders(array $headers = [])
     {
         foreach ($headers as $header) {
             if (($pos = strpos($header, ':')) !== false) {
-                $title = substr($header, 0, $pos);
-                $value = substr($header, $pos + 1);
+                $title = trim(substr($header, 0, $pos));
+                $value = trim(substr($header, $pos + 1));
 
-                $this->response->setHeader($title, $value);
+                if ($this->response instanceof Response) {
```

Comment:
```suggestion
                if ($this->response->hasHeader($title)) {
```

Reply:
We cannot remove `if ($this->response instanceof Response)` because the type of `$this->response` is `ResponseInterface`, and it does not have the `addHeader()` method.